backend/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+func uploadHandler(w http.ResponseWriter, r *http.Request) {
+	// Log the incoming request for debugging purpose
+	fmt.Println("Upload request received")
+
+	// Parse the form data
+	err := r.ParseMultipartForm(10 << 20) // Limit upload to 10MB
```

Comment:
There is a problem with using ParseMultipartForm - it will read the whole request body in memory and then parse.
for for a file of 100 MB - it will consume 100MB heap space in the program if 10 such request at the same time it will use 100 MB x 10 = 1 GB of heap space for just checking the file size.

Use 
- MaxBytesReader [https://pkg.go.dev/net/http#MaxBytesReader]
- MaxBytesError [https://pkg.go.dev/net/http#MaxBytesError]

`	r.Body = http.MaxBytesReader(w, r.Body, maxUploadSize)

	// Open destination file for writing
	filePath := "uploaded_file"
	dst, err := os.Create(filePath)
	if err != nil {
		http.Error(w, "Unable to create file", http.StatusInternalServerError)
		return
	}
	defer func() {
		dst.Close()
	}()

	// Stream file content to disk
	_, err = io.Copy(dst, r.Body)
	if err != nil {
		// Check for MaxBytesError
		var maxBytesErr *http.MaxBytesError
		if errors.As(err, &maxBytesErr) {
			http.Error(w, "File size exceeds the 100 MB limit", http.StatusRequestEntityTooLarge)
		} else {
			http.Error(w, "Failed to save file: "+err.Error(), http.StatusInternalServerError)
		}

		// Delete the partially written file
		os.Remove(filePath)
		return
	}
`